tests/integration/test_pg_store.py

```diff
@@ -419,3 +419,16 @@
 async def test_stop_no_connection(pg_job_store):
     pg_job_store.stop()
     # Well we didn't crash. Great.
+
+
+async def test_get_and_close_connection(pg_job_store):
+    conn1 = await pg_job_store.get_connection()
+    assert not conn1.closed
+    conn2 = await pg_job_store.get_connection()
+    assert conn2 is conn1
+    await pg_job_store.close_connection()
+    assert conn1.closed
+    conn2 = await pg_job_store.get_connection()
+    assert conn2 is not conn1
+    await pg_job_store.close_connection()
+    assert conn2.closed
```

Comment:
Why the 2 last lines ?
Do we get an error if we don’t close the connection ? In this case, I’d rather have them closed by a fixture so as to not make the test more complicated.

Also, this feels like 2 distinct tests put together (testing that when getting twice it’s the same object, testing that when getting when it’s closed, we get a new one), could we do this as 2 distinct tests ?
That point is debatable though, it’s integration tests, not unit tests, but I feel this is almost unit-test-like-ish. The only reason this is not unit test is that it needs IO. That being said, if we implement the other change above, it would nearly make sense to mock I/Os.

A debate to have :) It’s probably a bit of bike shedding but software design practices at the beginning of a project can be critical down the road.

Reply:
No I think you're right on all points. I'll do as you suggest it.

Reply:
As discussed I've splitted the test into two tests. And there's indeed no need to close the last connection, as it will be closed by the `pg_job_store` fixture.